js/viewer/annotation/base.js

```diff
@@ -2,35 +2,132 @@ function Base(attrs){
     this._attrs = {
-        "annotation-id" : attrs["annotation-id"] || -1,
+        // "annotation-id" : attrs["annotation-id"] || -1,
+        "graph-id" : attrs["graph-id"] || -1,
         "tag" : attrs["tag"] || "",
         "fill" : attrs["fill"] || "none",
         "stroke" :  attrs["stroke"] || "",
-        "stroke-width" : attrs["stroke-width"] || "",
-        "style" : ""
+        "stroke-width": attrs["stroke-width"] || this.constants.OUTPUT_SVG_STROKE_WIDTH,
+        "style" : "",
+        "vector-effect" : attrs["vector-effect"] || "non-scaling-stroke"
+    }
+    /**
+     * This functions checks to see if the svg component has valid attributes for it to be drawn. This function makes sure that no empty attribute is added.
+     * @param {object} attributes
+     * @return {boolean} Boolean values representing if the component has valid dimensions
+     */
+    this.hasDimensions = function(attributes) {
+        if ((attributes["height"] && attributes["width"]) || (attributes["d"]) || (attributes["cx"] && attributes["cy"] && attributes["r"])) {
+            return true;
+        }
+        return false;
+    }
+    /**
+     * This functions removes the style attributes, appends them to a string and returns them.
+     * @return {string}
+     */
+    this.getStyle = function() {
+        var styleString = "";
+        var styleAttributes = ['fill', "stroke", "stroke", "stroke-width"];
+
+        for (var i = 0; i <  styleAttributes.length; i++) {
+            if (this._attrs[styleAttributes[i]] && this._attrs[styleAttributes[i]].length > 0) {
+                styleString += styleAttributes[i] + ':' + this._attrs[styleAttributes[i]] + ';';
+                this._attrs[styleAttributes[i]] = null;
+            }
+        }
+
+        if (styleString[styleString.length - 1] == ';') {
+            styleString = styleString.slice(0, styleString.length - 1);
+        }
+
+        return styleString;
+    }
+
+    this.exportToSVG = function(){
+
+        // Check to see if there are necessary dimensions needed to construct the component. This makes sure that no empty components are added to the final SVG output file.
+        if (!this.hasDimensions(this._attrs)) {
+            return "";
+        }
+
+        var tag = this._attrs["tag"];
+        var rst = "<" + tag + " ";
+        var attr;
+
+        // if(this.parent.id !== ""){
+        //     rst += 'id="' + this.parent.id + '" ';
+        // };
+
+        var style = this.getStyle();
+        if (style.length > 0) {
+            this._attrs["style"] = style;
+        }
+
+        for(attr in this._attrs){
+            // Ingoring style because that will be added at the end.
+            // Ignoreing vector-effect becasue this properpty is not needed in the output file
+            if (!this._attrs[attr] || this._attrs[attr] === null || attr === "style" || attr == "vector-effect"){
+                continue;
+            }
+
+            switch(attr){
```

Comment:
The first two `case`s in this switch is the same as the `attr === 'style'` and `attr == "vector-effect"` that you added (both methods are just ignoring those attributes) Combine these so your code should be like this

```javascript
var ignoredAttributes = ["style", "vector-effect", "tag", "graph-id"];

for(attr in this._attrs){
  if (this._attrs[attr] == null || ignoredAttributes.indexOf(this._attrs[attr]) != -1) continue;

  rst += (attr + '="' + this._attrs[attr] + '" ');
}
```

You could make this more optimized but this is fine for now.